modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/BandMergeNode.java

```diff
@@ -172,8 +177,11 @@ protected GridCoverage execute() {
 
 					// color model
 					final ComponentColorModel cm = new ComponentColorModel(
-							ColorSpace.getInstance(ColorSpace.CS_LINEAR_RGB),
-							false, false, Transparency.OPAQUE,
+							(size == 1 && hasAlpha) ?
```

Comment:
Ouch, multiple inline conditional, that hurt my eyes. Could you use some temp variables instead? :-)

Reply:
wiil do.